### PR TITLE
[OPIK-448] [FE] UI on editing feedback score is slow

### PR DESCRIPTION
**Pull request by andriidudar**
## Details
Add the optimistic update for all cases when the user sets/ edits /deletes the feedback score

## Issues

Resolves #

### CODE REVIEW

File: apps/opik-frontend/src/api/traces/useTraceFeedbackScoreDeleteMutation.ts [hunk 1]
```diff
@@ -41,23 +55,51 @@ const useTraceFeedbackScoreDeleteMutation = () => {
...
+          },
+          deleteMutation,
+        );
+      } else {
+        // make optimistic update for traces
+        setTracesCache(queryClient, traceParams, deleteMutation);
+
+        // make optimistic update for trace
+        setTraceCache(queryClient, traceParams, deleteMutation);
+      }
```
Comment (ferc) [inline apps/opik-frontend/src/api/traces/useTraceFeedbackScoreDeleteMutation.ts:84]: Very clean code, love it!

File: apps/opik-frontend/src/api/traces/useTraceFeedbackScoreDeleteMutation.ts [hunk 2]
```diff
@@ -41,23 +55,51 @@ const useTraceFeedbackScoreDeleteMutation = () => {
         variant: "destructive",
       });
     },
+    onMutate: async (params: UseTraceFeedbackScoreDeleteMutationParams) => {
```
Comment (ferc) [inline apps/opik-frontend/src/api/traces/useTraceFeedbackScoreDeleteMutation.ts:58]: We have to add the `queryClient.cancelQueries(...)` step here -> https://tanstack.com/query/v5/docs/framework/react/guides/optimistic-updates#updating-a-single-todo

If not we could face some race conditions like:
1. You start the mutation `A` -> makes optimistic update (update cache) -> triggers request
2. Mutation `A` finishes and it's running the `onSettled` callback (refetch)
3. You make another mutation `B` -> makes optimistic update (`A` + `B` updates in the cache ✅ ) -> triggers request
4. Mutation `A` finishes the refetch -> updates the cache with `A` change (reverts last mutation from an old server-state) :x:
5. Eventually mutation `B` finishes the refetch -> updates the cache and `A` + `B` changes are in the cache ✅ 

So with cancellation of the queries step we remove the misleading old state of the step 4

**Review by ferc — APPROVED** [review]
Code looks very clean and it's great to have optimistic update for these mutations!

For next PR let's add the `.cancelQueries` call so we don't have to deal with some edge cases here

File: apps/opik-frontend/src/api/traces/useTraceFeedbackScoreDeleteMutation.ts (hunk 2 above)
Comment (andriidudar) [reply apps/opik-frontend/src/api/traces/useTraceFeedbackScoreDeleteMutation.ts:58]: Yes, you are right. Thanks for flagging this scenario. I have created a new PR with suggested improvements.

**Pull request merged**